scripts/sagas/session.js

```diff
@@ -0,0 +1,89 @@
+import { updatePath } from "redux-simple-router";
+import { call, take, fork, put } from "redux-saga/effects";
+
+import {
+  SESSION_SETUP,
+  SESSION_SERVERINFO_REQUEST,
+  SESSION_BUCKETS_REQUEST,
+  SESSION_LOGOUT
+} from "../constants";
+import * as notificationActions from "../actions/notifications";
+import * as sessionActions from "../actions/session";
+import { getClient, setupClient, resetClient } from "../client";
+
+
+export function* setupSession(session) {
+  setupClient(session);
+  yield put(notificationActions.clearNotifications({force: true}));
+  yield put(sessionActions.sessionBusy(true));
+  yield call(fetchServerInfo);
+  yield call(listBuckets);
+  yield put(sessionActions.setupComplete(session));
+  yield put(sessionActions.sessionBusy(false));
+}
+
+export function* sessionLogout() {
+  resetClient();
+  yield put(updatePath("/"));
+  yield put(notificationActions.notifySuccess("Logged out.", {persistent: true}));
+}
+
+export function* fetchServerInfo() {
+  const client = getClient();
+  try {
+    const serverInfo = yield call([client, client.fetchServerInfo]);
+    yield put(sessionActions.serverInfoSuccess(serverInfo));
+  } catch(error) {
+    yield put(notificationActions.notifyError(error));
+  }
+}
+
+export function* listBuckets() {
+  const client = getClient();
+  try {
+    // XXX We need to first issue a request to the "default" bucket in order
+    // to create user associated permissions, so we can access the list of
+    // buckets.
+    // ref https://github.com/Kinto/kinto/issues/454
+    const buck = client.bucket("default");
+    yield call([buck, buck.getAttributes]);
```

Comment:
Meanwhile added a check for default bucket information from serverInfo https://github.com/Kinto/kinto-admin/pull/77/commits/c180b2bab9fbc7215cd3d954353274769e001809